library/std/src/sync/once.rs

```diff
@@ -135,11 +38,13 @@ impl RefUnwindSafe for Once {}
 
 /// State yielded to [`Once::call_once_force()`]’s closure parameter. The state
 /// can be used to query the poison status of the [`Once`].
+// Uses `repr(transparent)` to make it possible to wrap a reference to the
+// inner `OnceState` so the dependency on the implementation is non-cyclic.
+// This is not a stable guarantee!
+#[repr(transparent)]
```

Comment:
This still shows up in docs, which encourages people to rely on it. Is there any way around this? If not, maybe #[cfg_attr(not(doc), repr(...))] might be better...

Reply:
I wanted to cleanly separate interface from implementation, but that is not really necessary.